modP2pImpl/src/org/aion/p2p/impl/TaskRequestActiveNodes.java

```diff
@@ -23,6 +23,7 @@ public TaskRequestActiveNodes(final IP2pMgr _mgr, final Logger p2pLOG) {
     public void run() {
         INode node = mgr.getRandom();
         if (node != null) {
+            Thread.currentThread().setName("p2p-reqNodes");
```

Comment:
since we have to set the thread's name, would it make sense for TaskRequestActiveNodes to extend Thread instead of implement Runnable?  I feel like we're assuming things about which thread this Runnable will be in, so why not just make it a Thread?

alternatively, make the thread-name-setting happen in P2pMgr line 231 instead (set the name before starting threadStatus)

Reply:
I think the purpose to use runnable is for saving the memory overhead, cause the task only executing `run()` method. you don't need to create a lot of objects by extending it to a Thread.

Reply:
Set the thead name mainly for more readable when you are using jstack to debug the thread info.